LanguageFeatures/Control-flow-collections/Null-aware-elements/warnings_A01_t01.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion As we do for other null-aware expressions like `?.` and `...?`,
+/// compilers and IDEs should probably warn if the inner expressions in
+/// null-aware elements are not potentially nullable since in that case, the `?`
+/// has no meaningful effect.
+///
+/// @description Check that it is a warning if the inner expressions in
+/// null-aware elements are not potentially nullable.
+/// @author sgrekhov22@gmail.com
+
+main() {
+  int e1 = 42;
+  int? e2 = 2 > 1 ? 2 : null;
+  int? e3 = 2 > 1 ? 3 : null;
+  int? e4 = 2 > 1 ? 4 : null;
+
+  if (e4 != null) {
+    <Object>{
+      ?e1,
+//    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+      ?e2!,
+//    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+      if (e3 != null)
+        ?e3
+//      ^
+// [analyzer] unspecified
+// [cfe] unspecified
+      else
+        ?e3, // Ok
+      ?e4
+//    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+    };
+
+    <Object>[
```

Comment:
We should probably put the set tests into a separate function `testSets` and the list tests into `testLists`, or something like that, in order to avoid any issues with flow-analysis (promotions) that arise in one test and changes the meaning of the next one.

Reply:
Moved to a separate function

Reply:
Thanks!